### PR TITLE
[CPU] reduce temporary weight memory of PA and SDPA

### PR DESCRIPTION
**Pull request by tiger100256-hu**
### Details:
 - *In the q @ k operation of PA and SDPA, need to use [q_len, s] @ [s, kv_len] =  [q_len, kv_len] memory to save the result. If there is multi batch size and multi heads, the memory shape is [B, H, q_len, kv_len], but in the implementation of CPU plugin, the operation will be split into multi tasks, every task only calculate a block size, so every task need memory [block_size, kv_len], the tasks will be scheduled to many different threads by tbb, so need to malloc [threads_num, block_size, kv_len] memory to save all the results. The temporary result will be used to do softmax operation and @ v in the same thread,  so it's safe to use [threads_num, block_size, kv_len] instead of [threads_num, H,  block_size, kv_len]*
 - *when loop_hk = true in PA, dim H became Hk, so the temporary weight memory shape became [threads_num, H / Hk,  block_size, kv_len]*
 - *about SDPA please refer to  https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html*

### Tickets:
 - *CVS-163700*

### CODE REVIEW

**Review by luo-cheng2021 — APPROVED** [review]
LGTM. Thanks.

**Comment by yuxu42** [comment]
Hi @maxnick Could you please review the PR? Thanks!

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/executor_pa.cpp [hunk 1]
```diff
@@ -1467,6 +1468,11 @@ struct MHAHelper {
         _weight.resize<float>({size_t{1}, size_t{1}, size_t{1}, size_t{1}});
     }
 
+    void resize_temporary_weight_buffer(const size_t& h) {
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/executor_pa.cpp:1471]: NIT. There is no need to pass "size_t" by const reference, it's better to pass it by value.

**Review by maxnick — APPROVED** [review]
